utils/__init__.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np
+from numpy.linalg import norm
+from numba import njit
+
+
+def grp_converter(groups, n_features):
+    """Create group partition and group indices.
+
+    Parameters
+    ----------
+    groups : int | list of ints | list of lists of ints
+        Partition of features used in the penalty on `w`.
+        If an int is passed, groups are contiguous blocks of features, of size
+        `groups`.
+        If a list of ints is passed, groups are assumed to be contiguous,
+        group number `g` being of size `groups[g]`.
+        If a list of lists of ints is passed, `groups[g]` contains the
+        feature indices of the group number `g`.
+
+    n_features : int
+        Number of features.
+
+    Returns
+    -------
+    grp_indices : array, shape (n_features,)
+        The group indices stacked contiguously
+        (e.g. [grp1_indices, grp2_indices, ...]).
+
+    grp_ptr : array, shape (n_groups + 1,)
```

Comment:
why putting these functions in the `__init__` submodule?

Reply:
Never mind, I didn't know where to put them (cf. https://github.com/scikit-learn-contrib/skglm/pull/123#discussion_r1025803593)